swim/memberlist.go

```diff
@@ -225,6 +230,31 @@ func (m *memberlist) MakeLeave(address string, incarnation int64) []Change {
 	return m.MakeChange(address, incarnation, Leave)
 }
 
+func (m *memberlist) MakeTombstone(address string, incarnation int64) []Change {
+	m.node.emit(MakeNodeStatusEvent{Tombstone})
+	return m.MakeChange(address, incarnation, Tombstone)
+}
+
+func (m *memberlist) Evict(address string) {
+	if m.local.Address == address {
+		// don't evict ourself, this is an assertion that no execution
+		// path should lead to this state
+		panic("can't evict self")
```

Comment:
Panic seems a bit extreme. Is there no other way to handle this?

It seems that in the attempt to "not evict" oneself, we are panicking which will exit! That will certainly lead to an eviction! :wink: